### PR TITLE
Add countries when adding an interaction - Part 1

### PR DESCRIPTION
**Pull request by web-bert**
## Description of change

**Hidden behind a feature flag called "interaction-add-countries"** so this PR should have no changes, unless this feature flag is present - this is to help make the PRs for this feature smaller.

To capture more up to date information about future countries of interest, countries currently exporting to or countries a company is not interested in exporting to, this adds a new question when adding an interaction for Export and Other "themes" to ask if any countries were discussed. If countries were discussed then three additional fields are revealed to ask for a list of countries for 

- Future countries of interest
- Countries currently exporting to
- Countries not interested in

Each of these fields is a Typeahead and there were a few modifications required to make the typeahead work in this scenario.

## Test instructions

Add a new interaction with a theme/type of "Export" or "Other" with any sub type selected, you should see a new question at the bottom asking "Were any countries discussed?"

_What should I see?_
 
### Before

Screenshot 2019-11-28 at 19 18 24

### After 

Screenshot 2019-11-28 at 19 18 34

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied. 
https://github.com/uktrade/data-hub-frontend/blob/develop/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to develop?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: test/functional/cypress/specs/interaction/add-interaction-spec.js [hunk 1]
```diff
@@ -243,32 +261,61 @@ describe('Adding interaction or service', () => {
...
+  cy.get(formSelectors.subService).select(subService)
+  cy.get(formSelectors.subject).type(subject)
+  cy.get(formSelectors.notes).type('Conversation with potential client')
+  cy.get(formSelectors.policyFeedbackNo).click()
+
+  checkCountriesDiscussed(formSelectors, visible, discussed)
 }
 
-const assertDetails = ({
+function assertDetails ({
```
Comment (rafenden) [inline test/functional/cypress/specs/interaction/add-interaction-spec.js]: We should have a convention on the usage of `function` vs `arrow func`.

File: test/functional/cypress/specs/interaction/add-interaction-spec.js [hunk 2]
```diff
@@ -7,194 +7,212 @@ const { companies, contacts, dashboard, investments } = require('../../../../../
...
       })
-    })
 
-    context('when in the context of a contact', () => {
-      beforeEach(() => {
-        cy.visit(contacts.interactions.createType(fixtures.default.id, 'export', 'service-delivery'))
-      })
+      context('when TAP service fields are empty', () => {
+        beforeEach(() => {
+          cy.visit(`/companies/${fixtures.default.id}/interactions/create/export/service-delivery`)
```
Comment (rafenden) [inline test/functional/cypress/specs/interaction/add-interaction-spec.js]: No `urls` module?

File: test/functional/cypress/specs/interaction/add-interaction-spec.js [hunk 3]
```diff
@@ -243,32 +261,61 @@ describe('Adding interaction or service', () => {
...
+  cy
+    .get(typeahead)
+    .click()
+    .get(textInput)
+    .type(text)
+    .type('{enter}')
+    .type('{esc}')
+}
+
+function checkCountriesDiscussed (formSelectors, visible, discussed) {
```
Comment (rafenden) [inline test/functional/cypress/specs/interaction/add-interaction-spec.js]: I would suggest `selectCountriesDiscussed` as `check` could be misunderstood as validation.

File: src/apps/interactions/controllers/edit.js [hunk 4]
```diff
@@ -38,56 +43,53 @@ async function buildForm (req, res, interactionId, validatedKind) {
     returnText: interactionId ? 'Return without saving' : 'Cancel',
     buttonText: interactionId
       ? 'Save and return'
-      : `Add ${lowerCase(validatedKind)}`,
+      : `Add ${lowerCase(kind)}`,
```
[the remark this answers is not in the document]
Comment (web-bert) [reply src/apps/interactions/controllers/edit.js]: I thought this was a good spot but according to the docs for `lowerCase` it "Converts string, as space separated words, to lower case."- so it does have to correct words on the button! Not exactly clear what is happening here though...

File: test/functional/cypress/specs/interaction/add-interaction-spec.js (hunk 1 above)
Comment (web-bert) [reply test/functional/cypress/specs/interaction/add-interaction-spec.js]: I agree, depending on what way you sway.... 😉

File: test/functional/cypress/specs/interaction/add-interaction-spec.js (hunk 2 above)
Comment (web-bert) [reply test/functional/cypress/specs/interaction/add-interaction-spec.js]: Missed this one, thanks

File: test/functional/cypress/specs/interaction/add-interaction-spec.js (hunk 3 above)
Comment (web-bert) [reply test/functional/cypress/specs/interaction/add-interaction-spec.js]: I have used the convention in the file of "populate"

**Pull request merged**